### ISSUE TITLE
[Effect Suggestion] Taxi!

### ISSUE THREAD

**Issue by iCeCoCaCoLa64** [gta-chaos-mod/ChaosModV#469]
**Name of effect**
I was thinking "Taxi!" or "Need a Ride?"

**Effect Description**
Spawns a taxi with a driver, then teleports the player into the backseat. Then they can use the taxi like normal and pick a destination if need be.

**Comment by EliRickard** [comment]
Not sure if it's possible to make it a useable taxi but we could certainly have it drive to a random set of locations like airport or military base etc

**Comment by anto4440** [comment]
Oh god
I just imagined an autopilot that couldnt be stopped until it reached its destination, which very well could be on the opposite side of the map.

**Comment by SeanMirrsen** [comment]
I am likewise uncertain that it's possible to spawn a taxi you can use like a regular taxi. It's also probably impossible to force the player to use said taxi. So I think for best/most chaotic results, this effect should do several things.

1. Replace current vehicle with a taxi. (per normal Replace Vehicle logic, so just put the player into a spawned taxi if they weren't in one)
2. Repeatedly check that the player is in a taxi, or any vehicle. Because replacing the vehicle is liable to cause mission failure and thus remove the vehicle.
3. For the duration of the effect, if the player is in a vehicle, activate autopilot with the taxi driver's AI. 
4. The destination should be either the current waypoint (mission waypoints included), a random mission blip if the player isn't on mission, or any map landmark, in that order of preference.

The primary difference from the Autopilot effect, would be that it doesn't cause the player to 'autopilot' on foot, and only serves to try and get the player to some destination.

**Comment by anto4440** [comment]
You can simply lock player in vehicle

**Comment by SeanMirrsen** [comment]
True. It comes down to what is possible through the script, then. If it's possible to actually force the player into a passenger seat of a spawned taxi, automatically select a destination, and lock the player in until the effect runs out, then it could work. 

Some mission scripts prevent taxis from working though, so there might be conflicts - again, depends on what the mod can accomplish here. It might still be preferable to just emulate a taxi, by spawning a taxi vehicle and autopiloting the player until the effect runs out, simply because it'll work in more circumstances.
